### PR TITLE
feat: Resource list enhancements

### PR DESCRIPTION
**Pull request by disq**
Implements #687 and https://github.com/cloudquery/cq-provider-aws/issues/879

~Maybe don't use "unlimited" globbing but allow only for class-level wildcards? ie. don't allow `s3.bu*` but only `s3.*`~

### CODE REVIEW

**Comment by bbernays** [comment]
I agree that we should only support blobs at the service level

File: pkg/core/fetch_test.go [hunk 1]
```diff
@@ -261,56 +260,106 @@ func Test_doNormalizeResources(t *testing.T) {
...
+			true,
+		},
+		{
+			"invalid glob 2",
+			[]string{"c1.*res1"},
+			nil,
+			map[string]*schema.Table{"c1.res1": nil, "c1.res2": nil, "c2.res3": nil, "c2.res4": nil},
+			nil,
+			true,
+		},
```
Comment (bbernays) [inline pkg/core/fetch_test.go:347]: The only other test I would add is a test to ensure `c1.res*` will fail

**Review by bbernays — APPROVED** [review]
LGTM other than the single comment

File: pkg/core/fetch.go [hunk 2]
```diff
@@ -418,44 +420,85 @@ func executeFetch(ctx context.Context, pLog zerolog.Logger, plugin plugin.Plugin
...
-// It replaces wildcard resource with all resources. Error is returned if:
+// doGlobResources returns a canonical list of resources given a list of requested and all known resources.
+// It replaces wildcard resource with all resources in non-wild mode. Error is returned if:
 //
 // * wildcard is present and other explicit resource is requested;
 // * one of explicitly requested resources is not present in all known;
 // * some resource is specified more than once (duplicate).
-func doNormalizeResources(requested []string, all map[string]*schema.Table) ([]string, diag.Diagnostics) {
-	if len(requested) == 1 && requested[0] == "*" {
+func doGlobResources(requested []string, allowWild bool, all map[string]*schema.Table) ([]string, diag.Diagnostics) {
```
Comment (roneli) [inline pkg/core/fetch.go:445]: nit: I think this function has multiple jobs, its better to have small functions rather then one big one.

- one to handle "*" 
- one to handle skip
- one to open glob patterns

**Review by roneli — COMMENTED** [review]
Looks great had a nit on the function as I think the functions should be small and with 1 job rather then do many jobs on the resource list, other than that the logic looks great.

File: pkg/core/fetch.go (hunk 2 above)
Comment (disq) [reply pkg/core/fetch.go:445]: skip is handled already in the same fn that's the point of normalize calling glob.
the glob pattern here is special and very basic so I don't think a new fn is needed just to wrap strings.HasSuffix and few exceptions.

Comment (disq) [reply pkg/core/fetch.go:445]: Updated

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**